configargparse.py

```diff
@@ -164,8 +164,25 @@ def parse(self, stream):
                 value = key_value_match.group("value")
 
                 if value.startswith("[") and value.endswith("]"):
-                    # handle special case of lists
-                    value = [elem.strip() for elem in value[1:-1].split(",")]
+                    l_value = value[1:-1]
+                    if l_value.startswith("[") and l_value.endswith("]"):
+                        # handle special case of list of lists
+                        # m = [['1', '2', '3'], ['4', '5', '6']]
+                        value = []
+                        while l_value and 0 < len(l_value):
```

Comment:
The initial way I implemented this was a hack, and this approach uses similar logic, so is even less robust. 
I'm going to replace this with json.loads which does proper parsing.